pyfiglet/__init__.py

```diff
@@ -135,30 +135,31 @@ def preloadFont(cls, font):
         """
         # Find a plausible looking font file.
         data = None
-        f = None
+        font_path = None
         for extension in ('tlf', 'flf'):
             fn = '%s.%s' % (font, extension)
             path = importlib.resources.files('pyfiglet.fonts').joinpath(fn)
             if path.exists():
-                f = path.open('rb')
+                font_path = path
                 break
             else:
                 for location in ("./", SHARED_DIRECTORY):
                     full_name = os.path.join(location, fn)
                     if os.path.isfile(full_name):
-                        f = open(full_name, 'rb')
+                        font_path = pathlib.Path(full_name)
```

Comment:
The CI should now succeed.

The fix was replacing the string typed `font_path` with `pathlib.Path` objects .  Line 141 already uses a `Path` object, which knows how to open files inside a Python `.egg` (where the test is taken fonts from), so we only needed to convert the string path in line 149 to a `Path` object so we can use its `.open` method in line 154.